app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt

```diff
@@ -0,0 +1,135 @@
+                Lecture("1003").apply {
+                    highlight = true
+                    changedTitle = true
+                },
+                Lecture("1004").apply {
+                    highlight = false
+                    changedTitle = true
+                })
+
+        private val oldLectures = listOf(
+                Lecture("2001").apply {
+                    highlight = false
+                    changedIsNew = false
+                },
+                Lecture("2002").apply {
+                    highlight = true
+                    changedIsNew = false
+                })
+
+        private val newLectures = listOf(
+                Lecture("2003").apply {
+                    highlight = true
+                    changedIsNew = true
+                },
+                Lecture("2004").apply {
+                    highlight = false
+                    changedIsNew = true
+                })
+
+        private val uncanceledLectures = listOf(
+                Lecture("3001").apply {
+                    highlight = false
+                    changedIsCanceled = false
+                },
+                Lecture("3002").apply {
+                    highlight = true
+                    changedIsCanceled = false
+                })
+
+        private val canceledLectures = listOf(
+                Lecture("3003").apply {
+                    highlight = true
+                    changedIsCanceled = true
+                },
+                Lecture("3004").apply {
+                    highlight = false
+                    changedIsCanceled = true
+                })
+
+    }
+
+    @Test
+    fun `getChangedLecturesCount returns 0 for an empty list`() {
+        val statistic = ChangeStatistic(emptyList())
+        assertThat(statistic.getChangedLecturesCount()).isEqualTo(0)
+    }
+
+    @Test
+    fun `getChangedLecturesCount returns the changed lectures count`() {
+        val statistic = ChangeStatistic(unchangedLectures + changedLectures)
+        assertThat(statistic.getChangedLecturesCount()).isEqualTo(2)
+    }
+
+    @Test
+    fun `getNewLecturesCount returns 0 for an empty list`() {
+        val statistic = ChangeStatistic(emptyList())
+        assertThat(statistic.getNewLecturesCount()).isEqualTo(0)
+    }
+
+    @Test
+    fun `getNewLecturesCount returns the new lectures count`() {
+        val statistic = ChangeStatistic(oldLectures + newLectures)
+        assertThat(statistic.getNewLecturesCount()).isEqualTo(2)
+    }
+
+    @Test
+    fun `getCanceledLecturesCount returns 0 for an empty list`() {
+        val statistic = ChangeStatistic(emptyList())
+        assertThat(statistic.getCanceledLecturesCount()).isEqualTo(0)
+    }
+
+    @Test
+    fun `getCanceledLecturesCount returns the canceled lectures count`() {
+        val statistic = ChangeStatistic(unchangedLectures + canceledLectures)
+        assertThat(statistic.getCanceledLecturesCount()).isEqualTo(2)
```

Comment:
you could use `.isEqualTo(canceledLectures.size())` as that would match the test description more precisely and is less fragile to changes, but still accurate. (same applies to other tests).

Reply:
Improved in e5337468b8d9380c77a4931e6dd748e7410eb429.